libfwupd/fwupd-client.c

```diff
@@ -449,6 +449,13 @@ fwupd_client_connect_get_proxy_cb (GObject *source,
 	g_autoptr(GError) error = NULL;
 	g_autoptr(GVariant) val = NULL;
 	g_autoptr(GVariant) val2 = NULL;
+	g_autoptr(GMutexLocker) locker = g_mutex_locker_new (&priv->proxy_mutex);
+
+	/* another thread did this for us */
+	if (priv->proxy != NULL) {
+		g_task_return_boolean (task, TRUE);
```

Comment:
You need to call `g_dbus_proxy_new_finish()` (and then drop the result) or the `g_dbus_proxy_new_async()` call will go unpaired.

Could do something like:
```
g_autoptr(GDBusProxy) proxy = NULL;
g_autoptr(GMutexLocker) locker = NULL;

proxy = g_dbus_proxy_new_finish (res, &error);

locker = g_mutex_locker_new (&priv->proxy_mutex);
if (priv->proxy != NULL)
  {
    g_task_return_boolean (task, TRUE);
    return;
  }
priv->proxy = g_steal_pointer (&proxy);

/* connect signals, etc. */
```

Reply:
> You need to call g_dbus_proxy_new_finish

Doh, of course. I've used your suggestion.